src/docfx/lib/ProcessUtility.cs

```diff
@@ -327,22 +326,19 @@ public static async Task RunInsideMutex(string mutexName, Func<Task> action)
         {
             Debug.Assert(!string.IsNullOrEmpty(mutexName));
 
-            // avoid the RunInsideMutex to be nested used
-            // doesn't support to require a lock before releasing a lock
-            // which may cause deadlock
-            if (t_innerCall.Value != null)
+            var lockPath = Path.Combine(AppData.MutexRoot, HashUtility.GetMd5Hash(mutexName));
+
+            // avoid the RunInsideMutex to be nested used with same mutex name
+            t_innerCall.Value = t_innerCall.Value ?? new HashSet<string>();
+            if (!t_innerCall.Value.Add(lockPath))
```

Comment:
put `Add` action into `try` block? since there is a `remove` in `finally` block

Reply:
ok!